stress-tests/utilities/utils.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+template <class F> struct y_combinator {
+    F f; // the lambda will be stored here
+
+    // a forwarding operator():
+    template <class... Args> decltype(auto) operator()(Args &&... args) const {
+        // we pass ourselves to f, then the arguments.
+        // [edit: Barry] pass in std::ref(*this) instead of *this
```

Comment:
silly comment. :) link directly to https://stackoverflow.com/questions/2067988/recursive-lambda-functions-in-c11 next to the struct instead?